roc/receiver.go

```diff
@@ -171,7 +171,12 @@ type Receiver struct {
 
 // Open a new receiver.
 // Allocates and initializes a new receiver, and attaches it to the context.
-func OpenReceiver(context *Context, config ReceiverConfig) (*Receiver, error) {
+func OpenReceiver(context *Context, config ReceiverConfig) (receiver *Receiver, err error) {
```

Comment:
Here and in most other places, you introduced named returns, but did not modify the code to use them. In result, defer logs variables that were never set. Please test your code when you're submitting PR for review.